core/camel-core-model/src/main/java/org/apache/camel/builder/RouteBuilder.java

```diff
@@ -665,7 +666,10 @@ protected void checkInitialized() throws Exception {
                         camelContext.adapt(ExtendedCamelContext.class).getErrorHandlerFactory());
             }
 
-            for (RouteBuilderLifecycleStrategy interceptor : lifecycleInterceptors) {
+            Collection<RouteBuilderLifecycleStrategy> strategies = new ArrayList<>(lifecycleInterceptors);
+            strategies.addAll(camelContext.getRegistry().findByType(RouteBuilderLifecycleStrategy.class));
+
```

Comment:
I wonder if we should sort these strategies, so a strategy can implement Ordered from Camel to influence if its run first or last etc.

Reply:
done